src/System.Security.Cryptography.Cng/tests/HandleTests.cs

```diff
@@ -36,6 +36,147 @@ public static void HandleDuplication()
                 Assert.Equal<byte>(propertyValue, actualValue);
             }
         }
+
+#if netstandard17
+        [Fact]
+        public static void SafeNCryptKeyHandle_ParentHandle_Invalid()
+        {
+            Assert.Throws<ArgumentNullException>(() => new SafeNCryptKeyHandle(IntPtr.Zero, null));
+
+            using (SafeHandle openButInvalid = new StateInformingSafeHandle(false))
+            using (SafeHandle closedAndInvalid = new StateInformingSafeHandle())
+            using (SafeHandle closedButValid = new StateInformingSafeHandle(true, false))
+            {
+                closedAndInvalid.Dispose();
+                closedButValid.Dispose();
+
+                // Preconditions.
+                Assert.False(openButInvalid.IsClosed, "openButInvalid.IsClosed");
+                Assert.True(openButInvalid.IsInvalid, "openButInvalid.IsInvalid");
+                Assert.True(closedButValid.IsClosed, "closedButValid.IsClosed");
+                Assert.False(closedButValid.IsInvalid, "closedButValid.IsInvalid");
+                Assert.True(closedAndInvalid.IsClosed, "closedAndInvalid.IsClosed");
+                Assert.True(closedAndInvalid.IsInvalid, "closedAndInvalid.IsInvalid");
+
+                // Tests
+                Assert.Throws<ArgumentException>(() => new SafeNCryptKeyHandle(IntPtr.Zero, openButInvalid));
+                Assert.Throws<ArgumentException>(() => new SafeNCryptKeyHandle(IntPtr.Zero, closedButValid));
+                Assert.Throws<ArgumentException>(() => new SafeNCryptKeyHandle(IntPtr.Zero, closedAndInvalid));
+            }
+        }
+
+        [Fact]
+        public static void SafeNCryptKeyHandle_InvalidKey_ParentHandle_NotKeptAlive()
+        {
+            using (SafeHandle parentHandle = new StateInformingSafeHandle())
+            {
```

Comment:
Clearly was copied from a case where there were asserts after the inner using.  Tidied it up.